src/iosrv/srv.c

```diff
@@ -643,10 +653,20 @@ dss_start_one_xstream(hwloc_cpuset_t cpus, int xs_id)
 			       dss_srv_handler, dx, attr,
 			       &dx->dx_progress);
 	if (rc != ABT_SUCCESS) {
-		D_ERROR("create xstream failed: %d\n", rc);
+		D_ERROR("create progress ULT failed: %d\n", rc);
 		D_GOTO(out_xstream, rc = dss_abterr2der(rc));
 	}
 
+	if (dx->dx_main_xs) {
+		rc = ABT_thread_create(dx->dx_pools[DSS_POOL_SHARE],
+				       dss_gc_ult, NULL,
+				       ABT_THREAD_ATTR_NULL, NULL);
```

Comment:
I'm afraid that in current implementation, the proper place to create this ULT should be in dss_srv_handler(). because we do waiting all ULTs exit, init/fini TLS there.

Reply:
ah, right